pkg/instance/instance.go

```diff
@@ -867,181 +785,32 @@ func (i *Instance) SetStartupProbe(startupProbe *v1.Probe) error {
 
 // AddSidecar adds a sidecar to the instance
 // This function can only be called in the state 'Preparing' or 'Committed'
-func (i *Instance) AddSidecar(sidecar *Instance) error {
-
-	if !i.IsInState(Preparing, Committed) {
-		return ErrAddingSidecarNotAllowed.WithParams(i.state.String())
-	}
-	if sidecar == nil {
+func (i *Instance) AddSidecar(ctx context.Context, sc SidecarManager) error {
```

Comment:
Now, the dev needs to understand what a Kubernetes sidecar is when adding obsy, bittwister, or tshark. The knowledge we assume from the devs should be as minimal as possible. Before the dev was just adding the features and under the hood, knuu added these as a sidecar.
Any ideas on how we can retain that property?

Reply:
Maybe we can call it a different name as it is kinda pluggin for knuu and users can develop their own sidecars if they want. However, I also think that users do not need to know alot about sidecars, maybe what they understand from knuu sidecar is much more simplified than k8s sidecar as there is a sidecar interface with a couple of funcs that they need to implement. I am ok with calling it a different name. Do you have any proposal how to make it more friendly to the users, not only renaming it, in general?

Reply:
I think moving these functions to a separate object is a significant improvement to the status quo.

I came up with these alternative names:
- Capability
- Feature
- Ability
- Module

But maybe it is fine to keep the sidecar term, as you said, the knuu sidecar is much more simplified than the k8s sidecar.

Would like to have input from @tty47

Reply:
I like the name `Module` more than others

Reply:
mm I love the `Plugin` name more.